src/client.cpp

```diff
@@ -400,7 +400,9 @@ void Client::step(float dtime)
 	// Control local player (0ms)
 	LocalPlayer *player = m_env.getLocalPlayer();
 	assert(player);
-	player->applyControl(dtime, &m_env);
+	PlayerSettings settings;
```

Comment:
reading settigns on each client step is crazy, can that be part of client constructor ?

Reply:
Remember that things like `free_move` and `noclip` are handled via settings. If you check the original movement functions, these are riddled with `g_settings->getBool(…)` calls:
https://github.com/minetest/minetest/blob/master/src/localplayer.cpp#L491 ff

Not to mention that these functions check(ed) the same settings several times, see https://github.com/minetest/minetest/blob/master/src/localplayer.cpp#L505 just a few lines below.

Reply:
Please see #7243 for a settings caching mechanism. These would then be part of the `LocalPlayer` and would behave as requested.

Reply:
Fixed due to the merge of #7243